### DISCUSSION TITLE
Easily disable X or Y axis scroll

### DISCUSSION THREAD

**Discussion by blitzve0** [idiotWu/smooth-scrollbar#397] [Ideas]
A simple control like x: false and y: true would be  a life saver as right now just to disable x axis scroll we have to create plugins with delta[x]=0 which works fine but it would be so much better to have this in the configuration

**Comment by dolphin-wood** [comment]
[This plugin](https://github.com/idiotWu/smooth-scrollbar/discussions/357#discussioncomment-854036) will be added to the official plugins repository in the next version. All you need to do is to import & register the plugin by then:

```js
// smooth-scrollbar@9
import DisableScrollPlugin from '@smooth-scrollbar/disable-scroll-plugin';

new Scrollbar(elem, {
  plugins: [new DisableScrollPlugin({ x: true })]
})
```

**Reply by blitzve0** [reply]
That is Awesome

**Comment by IrynaVynnychenko** [comment]
How to add this plugin to all?

This variant:
```
<section data-scrollbar>
  
</section>
```

<script> Scrollbar.initAll(); </script>

**Reply by dolphin-wood** [reply]
The above comment was just a proposal for the next version(v9). For our current version(v8.x), you can check this implementation: https://github.com/idiotWu/smooth-scrollbar/discussions/357#discussioncomment-854036
